File: src/Connector.js
import React, {Component} from 'react';
import CommentBox from "./commentbox/CommentBox";
import Notification from "./notification/Notification";

class Connector extends Component {
    constructor() {
        super();
        this.state = {
            noOfComments: 0
        };

        this.updateNoOfComments = this.updateNoOfComments.bind(this)

    }
    render() {
        return (
            <div className="Connector">
                <CommentBox updateNoOfComments={this.updateNoOfComments}/>
                <Notification noOfComments={this.state.noOfComments}/>
            </div>
        );
    }

    updateNoOfComments(commetsNumber) {
      this.setState({noOfComments: commetsNumber})
    }
}

export default Connector;

File: src/main/scala/samples/CommentListExample/connector/Connector.scala
package samples.CommentListExample.connector

import com.github.ahnfelt.react4s._
import samples.CommentListExample.Notification.Notification
import samples.CommentListExample.commentList.CommentBox
import samples.CommentListExample.commentList.CommentBox.UpdateNoOfComments

case class Connector() extends Component[NoEmit] {

  val noOfCommentsS = State(0)

  override def render(get: Get): ElementOrComponent = {

    def handleCommentBoxEvents(msg: CommentBox.Msg): Unit = msg match {
      case UpdateNoOfComments(noOfComments) => noOfCommentsS.set(noOfComments)
    }

    E.div(
      E.div(
        Component(CommentBox).withHandler(handleCommentBoxEvents)
      ),
      E.div(
        Component(Notification, get(noOfCommentsS))
      )
    )
  }
}
